server/odc-service/src/main/java/com/oceanbase/odc/service/db/schema/DBSchemaIndexService.java

```diff
@@ -0,0 +1,207 @@
+import org.springframework.validation.annotation.Validated;
+
+import com.oceanbase.odc.core.authority.util.SkipAuthorize;
+import com.oceanbase.odc.core.shared.constant.OrganizationType;
+import com.oceanbase.odc.core.shared.constant.ResourceRoleName;
+import com.oceanbase.odc.core.shared.constant.ResourceType;
+import com.oceanbase.odc.core.shared.exception.NotFoundException;
+import com.oceanbase.odc.core.shared.exception.NotImplementedException;
+import com.oceanbase.odc.metadb.dbobject.DBColumnEntity;
+import com.oceanbase.odc.metadb.dbobject.DBColumnRepository;
+import com.oceanbase.odc.metadb.dbobject.DBObjectEntity;
+import com.oceanbase.odc.metadb.dbobject.DBObjectRepository;
+import com.oceanbase.odc.service.collaboration.project.ProjectService;
+import com.oceanbase.odc.service.connection.ConnectionService;
+import com.oceanbase.odc.service.connection.database.DatabaseService;
+import com.oceanbase.odc.service.connection.database.model.Database;
+import com.oceanbase.odc.service.connection.model.ConnectionConfig;
+import com.oceanbase.odc.service.db.schema.model.OdcDBColumn;
+import com.oceanbase.odc.service.db.schema.model.OdcDBObject;
+import com.oceanbase.odc.service.db.schema.model.QueryDBObjectParams;
+import com.oceanbase.odc.service.db.schema.model.QueryDBObjectResp;
+import com.oceanbase.odc.service.db.schema.model.SyncDBObjectReq;
+import com.oceanbase.odc.service.iam.ProjectPermissionValidator;
+import com.oceanbase.odc.service.iam.auth.AuthenticationFacade;
+import com.oceanbase.tools.dbbrowser.model.DBObjectType;
+
+import lombok.NonNull;
+
+/**
+ * @author gaoda.xy
+ * @date 2024/3/28 13:41
+ */
+@Service
+@Validated
+@SkipAuthorize("permission check inside")
+public class DBSchemaIndexService {
+
+    @Autowired
+    private ProjectPermissionValidator projectPermissionValidator;
+    @Autowired
+    private DatabaseService databaseService;
+    @Autowired
+    private ProjectService projectService;
+    @Autowired
+    private AuthenticationFacade authenticationFacade;
+    @Autowired
+    private ConnectionService connectionService;
+    @Autowired
+    private DBColumnRepository dbColumnRepository;
+    @Autowired
+    private DBObjectRepository dbObjectRepository;
+
+    public QueryDBObjectResp listDatabaseObjects(@NonNull @Valid QueryDBObjectParams params) {
+        QueryDBObjectResp resp = new QueryDBObjectResp();
+        List<Long> queryDatabaseIds = new ArrayList<>();
+        if (params.getProjectId() != null && params.getDatasourceId() != null) {
+            throw new IllegalArgumentException("projectId and datasourceId cannot be set at the same time");
+        }
+        if (params.getProjectId() != null) {
+            projectPermissionValidator.checkProjectRole(params.getProjectId(), ResourceRoleName.all());
+            if (CollectionUtils.isNotEmpty(params.getDatabaseIds())) {
+                List<Database> databases = databaseService.listDatabasesByIds(params.getDatabaseIds());
```

Comment:
is project info will be full filled by `listDatabasesByIds`?

Reply:
No, only filled by project id, and we only need project id.